tensorflow_addons/utils/resource_loader.py

```diff
@@ -46,5 +52,54 @@ def __init__(self, relative_path):
     @property
     def ops(self):
         if self._ops is None:
+            self.display_warning_if_incompatible()
             self._ops = tf.load_op_library(get_path_to_datafile(self.relative_path))
         return self._ops
+
+    def display_warning_if_incompatible(self):
+        global abi_warning_already_raised
+        if abi_is_compatible() or abi_warning_already_raised:
+            return
+
+        warnings.warn(
+            "You are currently using TensorFlow {} and trying to load a custom op ({})."
+            "\n"
+            "TensorFlow Addons has compiled its custom ops against TensorFlow {}, "
+            "and there are no compatibility guarantees between the two versions. "
+            "\n"
+            "This means that you might get segfaults when loading the custom op, "
+            "or other kind of low-level errors.\n If you do, do not file an issue "
+            "on Github. This is a known limitation."
+            "\n\n"
+            "It might help you to fallback to pure Python "
+            "ops with TF_ADDONS_PY_OPS . To do that, see "
+            "https://github.com/tensorflow/addons#gpucpu-custom-ops "
+            "\n\n"
+            "You can also change the TensorFlow version installed on your system. "
+            "You would need a TensorFlow version equal to or above {} and strictly "
```

Comment:
Can you add "installed from pip" somewhere in there. I suspect that is just as likely of an issue as incompatible version

Reply:
I'll add it. But this warning won't appear if someone is building from source. I don't know if there is any way we can detect that. If TF was exposing an API to get info about the build, we could use that. As far as I know, such an API doesn't exist.